python_modules/dagster/dagster/_core/host_representation/external_data.py

```diff
@@ -1633,8 +1637,10 @@ def external_resource_data_from_def(
         ),
     ):
         resource_type = _get_class_name(resource_type_def.configurable_resource_cls)
+        dagster_maintained = resource_type_def._dagster_maintained
```

Comment:
this can probably be pulled out of the if/else and simplified to

`dagster_maintained = resource_type_def._dagster_maintained._dagster_maintained`

since I think that `resource_type_def` is guaranteed to be a ResourceDefinition (and is already the inner, non-wrapped resource in the case of ResourceWithKeyMapping)